cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/CombinedHBaseMetricsTable.java

```
@@ -106,19 +136,68 @@ public long incrementAndGet(byte[] row, byte[] column, long delta) {
   @Override
   public void delete(byte[] row, byte[][] columns) {
     v3HBaseTable.delete(row, columns);
-    v2HBaseTable.delete(row, columns);
+    if (v2HBaseTable != null) {
+      try {
+        v2HBaseTable.delete(row, columns);
+      } catch (Exception e) {
+        handleTableException(e);
+      }
+    }
+  }
+
+  private void handleTableException(Exception e) {
+    if (e instanceof IOException) {
+      if (!v2MetricsTableExists()) {
+        Closeables.closeQuietly(v2HBaseTable);
+        v2HBaseTable = null;
+      }
+    }
```

otherwise, I believe you still want to rethrow the original exception?